### PR TITLE
MRG, MAINT: Bump minimums

### PR DESCRIPTION
**Pull request by larsoner**
Remove some cruft by bumping minimums that will be > 2 years old by the time we next release (March 2021).

Not sure if this or #7955 should go in first (probably #7955 since the rebase there would be more painful).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -33,8 +33,7 @@ matrix:
         # Old dependencies
         - os: linux
           env: PYTHON_VERSION=3.6
-               CONDA_DEPENDENCIES="numpy=1.13 scipy=1.0 matplotlib=2.1 pandas=0.21 scikit-learn=0.19"
-               CONDA_CHANNELS="conda-forge"
+               CONDA_DEPENDENCIES="numpy=1.15.2 scipy=1.1 matplotlib=3.0.3 pandas=0.23.4 scikit-learn=0.20.2"
```
Comment (agramfort) [inline .travis.yml:36]:
```suggestion
               CONDA_DEPENDENCIES="numpy=1.15.4 scipy=1.1 matplotlib=3.0.3 pandas=0.23.4 scikit-learn=0.20.2"
```

**Review by agramfort — COMMENTED** [review]
@drammock ok for you?

**Comment by drammock** [comment]
> @drammock ok for you?

Yes, although I agree with @larsoner that #7955 should get merged before this one.

File: README.rst [hunk 2]
```diff
@@ -86,23 +86,22 @@ Dependencies
 The minimum required dependencies to run MNE-Python are:
 
 - Python >= 3.6
-- NumPy >= 1.13.3
-- SciPy >= 1.0.0
+- NumPy >= 1.15.4
```
Comment (agramfort) [inline README.rst:89]: you write 1.15.2 in travis. See comment above

**Review by agramfort — APPROVED** [review]

**Review by drammock — APPROVED** [review]
assuming CIs are happy LGTM

**Pull request merged**